Katello/katello
# Fixes #35091 - Python content isn't synced to smart proxies

#### What are the changes introduced in this pull request?
Updates pulp-python to 3.7.1.  Also adds a fix where updating Python repositories fails because nil isn't accepted for the update api.

#### Considerations taken when implementing this change?
I considered just calculating the packages to include, but that would be difficult to do in a generic way (and Python is a generic type).

#### What are the testing steps for this pull request?
1) Sync Python content to a smart proxy. Notice that the distributed Python content has no packages.
2) Upgrade pulp-python to 3.7.1 + `bundle update`.
3) Sync again (complete).
4) See that there is Python content being served now on the smart proxy.
5) Try updating a Python repository to have empty includes, excludes, and package types.
